Pod/Classes/WPMediaCollectionViewCell.m

```diff
@@ -148,6 +150,44 @@ - (void)displayAudioAssetTypePlaceholder
     [self setCaption:[self stringFromTimeInterval:audioDuration]];
 }
 
+- (void)displayVideoAssetTypePlaceholder
+{
+    self.placeholderStackView.hidden = NO;
+    self.imageView.hidden = YES;
+
+    if ([self.asset respondsToSelector:@selector(fileExtension)]) {
+        NSString *extension = [[self.asset fileExtension] uppercaseString];
+        self.documentExtensionLabel.text = extension;
+    } else {
+        self.documentExtensionLabel.text = NSLocalizedString(@"VIDEO", @"Label displayed on audio media items.");
+    }
+
+    self.placeholderImageView.image = [[WPMediaPickerResources imageNamed:@"gridicons-video-camera" withExtension:@"png"] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
+
+    NSTimeInterval duration = [self.asset duration];
+    [self setCaption:[self stringFromTimeInterval:duration]];
+}
+
+- (void)displayImageAssetTypePlaceholder
+{
+    self.placeholderStackView.hidden = NO;
+    self.imageView.hidden = YES;
+
+    if ([self.asset respondsToSelector:@selector(fileExtension)]) {
```

Comment:
Perhaps we could extract this to some kind of `showAssetFileExtensionWithDefault:` method or something?